src/core/containers/error-simulation/utils.js

```diff
@@ -0,0 +1,14 @@
+/* eslint-disable react/prop-types */
+import defaultConfig from 'config';
+import React from 'react';
+
+import NotFound from 'core/components/ErrorPage/NotFound';
+
+export const render404WhenNotAllowed = (Component) => ({
+  config = defaultConfig, ...props
+}) => {
+  if (!config.get('allowErrorSimulation')) {
```

Comment:
This is a cool idea but I think it should take an argument that gets passed to the `config.get` call; then it's an actual utility function and not specific to this one config call. Because we could use this for the other error page routes only active in development mode.

```jsx
export const render404Unless = (configToCheck) => (Component) => [...]

if (config.get(configToCheck) {
  return <Component />
}

return <NotFound />
```
Type of thing.